modules/dnn_superres/src/dnn_superres.cpp

```diff
@@ -0,0 +1,385 @@
+                }
+                else
+                {
+                    //
+                }
+            }
+            else
+            {
+                std::cout << "Model not specified. Please set model via setModel(). \n";
+            }
+        }
+
+        void DnnSuperResImpl::upsampleMultioutput(Mat img, std::vector<Mat> &imgs_new, std::vector<int> scale_factors, std::vector<String> node_names)
+        {
+            CV_Assert(scale_factors.size() == node_names.size());
+            CV_Assert(!scale_factors.empty());
+            CV_Assert(!node_names.empty());
+
+            if ( this->alg != "lapsrn" )
+            {
+                std::cout << "Only LapSRN support multiscale upsampling for now!" << std::endl;
+                return;
+            }
+
+            if( !net.empty() )
+            {
+                if ( this->alg == "lapsrn" )
+                {
+                    Mat orig = img;
+
+                    //Preprocess the image: convert to YCrCb float image and normalize
+                    Mat preproc_img;
+                    preprocess_YCrCb(orig, preproc_img);
+
+                    //Split the image: only the Y channel is used for inference
+                    Mat ycbcr_channels[3];
+                    split(preproc_img, ycbcr_channels);
+
+                    Mat Y = ycbcr_channels[0];
+
+                    //Create blob from image so it has size 1,1,Width,Height
+                    cv::Mat blob;
+                    dnn::blobFromImage(Y, blob, 1.0);
+
+                    //Get the HR outputs
+                    std::vector <Mat> outputs_blobs;
+                    this->net.setInput(blob);
+                    this->net.forward(outputs_blobs, node_names);
+
+                    for(unsigned int i = 0; i < scale_factors.size(); i++)
+                    {
+                        std::vector <Mat> model_outs;
+                        dnn::imagesFromBlob(outputs_blobs[i], model_outs);
+                        Mat out_img = model_outs[0];
+                        Mat reconstructed;
+
+                        reconstruct_YCrCb(out_img, preproc_img, reconstructed, scale_factors[i]);
+
+                        imgs_new.push_back(reconstructed);
+                    }
+                }
+            }
+            else
+            {
+                std::cout << "Model not specified. Please set model via setModel(). \n";
+            }
+        }
+
+
+        void DnnSuperResImpl::upsampleVideo(String inputPath, String outputPath)
+        {
+            VideoCapture inputVideo(inputPath);
+            int ex = static_cast<int>(inputVideo.get(CAP_PROP_FOURCC));
+            Size S = Size((int) inputVideo.get(CAP_PROP_FRAME_WIDTH) * this->sc,
+                            (int) inputVideo.get(CAP_PROP_FRAME_HEIGHT) * this->sc);
+
+            VideoWriter outputVideo;
+            outputVideo.open(outputPath, ex, inputVideo.get(CAP_PROP_FPS), S, true);
```

Comment:
This code of writer creation doesn't work well.

Do we really need this method in the module code? What is about cameras?
Does it make sense? Video compression algorithms can significantly amend results of algorithms.

Reply:
Removed it from part of the module into just sample code, it makes more sense there.